frontend/src/components/verticalPodAutoscaler/Details.tsx

```diff
@@ -0,0 +1,199 @@
+                        let maxAllowed = '';
+                        if (item.maxAllowed && Object.keys(item.maxAllowed).length > 0) {
+                          Object.keys(item.maxAllowed).map(key => {
+                            maxAllowed += `${item.maxAllowed[key]},`;
+                          });
+                        }
+                        maxAllowed = maxAllowed.slice(0, -1);
+                        return maxAllowed;
+                      },
+                    },
+                    {
+                      label: t('translation|Mode'),
+                      getter: item => item?.mode,
+                    },
+                  ]}
+                />
+              </>
+            ),
+          },
+          {
+            id: 'headlamp.vpa-status-conditions',
+            section: (
+              <>
+                <SectionHeader title={t('translation|Conditions')} />
+                <SimpleTable
+                  data={item.status.conditions}
+                  columns={[
+                    {
+                      label: t('translation|Type'),
+                      getter: item => item.type,
+                    },
+                    {
+                      label: t('translation|Status'),
+                      getter: item => item.status,
+                    },
+                    {
+                      label: t('translation|Reason'),
+                      getter: item => item.reason,
+                    },
+                    {
+                      label: t('translation|Message'),
+                      getter: item => item.message,
+                    },
+                    {
+                      label: t('translation|Last Transition Time'),
+                      getter: item => <DateLabel date={item.lastTransitionTime} format="brief" />,
+                    },
+                  ]}
+                />
+              </>
+            ),
+          },
+          {
+            id: 'headlamp.vpa-recommendations',
+            section: (
+              <>
+                <SectionHeader title={t('translation|Recommendations')} />
+                <SimpleTable
+                  data={item.status.recommendation.containerRecommendations}
+                  columns={[
+                    {
+                      label: t('glossary|Container'),
+                      getter: item => item.containerName,
+                    },
+                    {
+                      label: t('glossary|Lower Bound'),
+                      getter: item => {
+                        let lowerBound = '';
+                        if (item.lowerBound && Object.keys(item.lowerBound).length > 0) {
+                          Object.keys(item.lowerBound).map(key => {
+                            lowerBound += `${key}:${item.lowerBound[key]},`;
+                          });
+                        }
+                        lowerBound = lowerBound.slice(0, -1);
+                        return lowerBound;
```

Comment:
This is the rest of the similar code could be abstracted into a helper function instead of having the code repeated.